### ISSUE TITLE
为什么yolo11n这么小的模型，推理占用的显存却很大？

### ISSUE THREAD

**Issue by TonyUSTC** [ultralytics/ultralytics#18635]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

yolo11n只有5M，推理时占用1G的显存，怎么降低呢？尝试半精度half，缩小imgsz，没有用。
results = model(img,imgsz=160,half=True)

**Comment by UltralyticsAssistant** [comment]
👋 Hello @TonyUSTC, thank you for your interest in Ultralytics 🚀! We recommend checking out the [Docs](https://docs.ultralytics.com) where you'll find detailed guidance and usage examples for [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/). Many commonly asked questions are addressed there.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) so we can better understand and debug the issue you’re describing.

If this is a custom training or inference ❓ Question, please include as much detail as possible, such as the settings you are using, specific hardware information, and complete logs or code snippets. For reference, make sure you’re following best practices outlined in our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

### Debugging Steps

If your issue is related to high inference memory usage, here are a few suggestions you can try as you investigate:
1. Verify you’re running the latest version of Ultralytics by upgrading with:
```bash
pip install -U ultralytics
```
2. Ensure your environment meets all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) and uses [**Python>=3.8**](https://www.python.org/) and [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/).
3. Experiment with precision settings (`float16`, `float32`) and batch sizes to see how they affect memory usage.
4. Check the shape and size of your input images to ensure they are consistent with the expected model input size.

### Environments

For easy experimentation, YOLO is verified to run in the following environments with full dependencies preinstalled:
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

### Community Support

Join the Ultralytics community for additional insights and discussions:
- [Discord](https://discord.com/invite/ultralytics) 🎧
- [Discourse](https://community.ultralytics.com)
- [Subreddit](https://reddit.com/r/Ultralytics)

### CI Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are successfully passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

This is an automated response to help get you started. An Ultralytics engineer will review your question and assist you further shortly. Thank you for your patience! 😊

**Comment by Kayzwer** [comment]
@TonyUSTC 
```
from ultralytics import YOLO


model = YOLO("yolo11n.pt")
model("./ultralytics/assets/bus.jpg", imgsz=160, half=True, device=0)
```

my gpu is gtx 1050 mobile, I complete the inference with only 0.147gb of ram, try specify device=0

**Comment by glenn-jocher** [comment]
@TonyUSTC it seems like specifying `device=0` didn't resolve your issue. The increased memory usage might be due to CUDA context initialization overhead or other GPU-specific factors. You can test running inference on the CPU (`device='cpu'`) to compare memory usage. Additionally, ensure your GPU drivers and PyTorch version are up to date. If the problem persists, please share more details about your environment for further assistance.
